bundles/org.smarthomej.binding.deconz/src/main/java/org/smarthomej/binding/deconz/internal/netutils/WebSocketConnection.java

```diff
@@ -76,6 +76,10 @@ public WebSocketConnection(WebSocketConnectionListener listener, WebSocketClient
         this.watchdogInterval = watchdogInterval;
     }
 
+    public void setWatchdogInterval(int watchdogInterval) {
+        this.watchdogInterval = watchdogInterval;
+    }
+
     public void start(String ip) {
```

Comment:
I am missing a line of code which sets the `connectionState` to `ConnectionState.CONNECTING`. Do you think that is the reason for multiple tries to start the websocket?

Reply:
Yes, that's the reason why it tries to start multiple times. I fixed it. Nevertheless the multiple calls to `connectionListener.webSocketConnectionLost(reason)` would still create multiple new instances on a single connection loss. This is the result of the call in l. 101/106 is that sometimes the `@OnWebsocketClose` is not called, it depends on when the connection attempt fails. If it is called, the bridge handler is notified two times. If we omit the call here, we could miss some failed connection attempts. Unfortunately it is not clearly documented when `@OnWebsocketClose` is called.